public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -130,7 +124,20 @@ export class PrometheusDatasource {
         }
       });
 
+      var tableResponses = _.filter(responseList, (response, index) => {
```

Comment:
This code looks a bit messy and feels hacky (the way time series data is handled differently, and table data is merged and then transformed). Wonder if there is a nicer way to process response. Also was there a reason for removing the response.status error handling? Has Prometheus changed how it returns errors?

Reply:
> Also was there a reason for removing the response.status error handling? Has Prometheus changed how it returns errors?
Sorry, it is mistake. revert the change.

Reply:
> Wonder if there is a nicer way to process response.

Squashing all time series to only one series, is easiest way.

The table panel doesn't distinguish time series, it convert one datapoint to one table record.
So, I think it is not hacky.
If calling transformMetricDataToTable() for each time series, we need to care about which label key is missing.
find all possible label key before calling transformMetricDataToTable(), and then pass it to transformMetricDataToTable().
You might feel it is messy too...